### PR TITLE
feat(pkger): add metrics svc middleware

### PR DESCRIPTION
**Pull request by jsteenb2**
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: kit/metric/client.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package metric
```
Comment (kelwang) [inline kit/metric/client.go]: Is there any special reason to create a new folder under kit? we have a prometheus package under the root. Or we can make a package inside the pkgr folder?

File: kit/metric/client.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+// Record returns a record fn that is called on any given return err. If an error is encountered
+// it will register the err metric. The err is never altered.
+func (c *REDClient) Record(method string) func(error) error {
+	start := time.Now()
+	return func(err error) error {
+		c.reqs.With(prometheus.Labels{"method": method})
+
+		if err != nil {
+			code := "unknown"
```
Comment (kelwang) [inline kit/metric/client.go]: if you can ErrorCode() later, for a non wrapped error, it will always return internal

Comment (jsteenb2) [reply kit/metric/client.go]: this coudl be reached if there is a `Error` type that doesn't set the code, which happens fairly frequently in our codebase unfortunately

File: kit/metric/client.go (hunk 1 above)
Comment (jsteenb2) [reply kit/metric/client.go]: great question, here's the TLDR I gave @goller  in slack about this:

> yah, was going back and forth on that. we definittely could add it to prom. I decided for a new one for 2 reasons.
> 1. metric is dead obvious for someone who looks at the pkg name (“It probably does metrics…“)
> 2. wasn’t sure how tied we are to prom, if we end up using some other metrics engine, its a dependency issue that no one has to worry about except the metric pkg (although dependency change would force updates elsewhere)

File: kit/metric/client.go (hunk 2 above)
Comment (jsteenb2) [reply kit/metric/client.go]: I take that back, you're right, its not reachable 🤦‍♂, inlined it 👍

**Pull request merged**